play/src/front/Connection/RoomConnection.ts

```diff
@@ -409,6 +413,60 @@ export class RoomConnection implements RoomConnection {
+                                iframeListener.sendLeaveMucEventToChatIframe(`${scene.roomUrl}/${slugify(name)}`);
+                                chatZoneLiveStore.set(false);
+                                break;
+                            }
+                            case "muteMicrophoneSpaceUserMessage": {
+                                if (subMessage.muteMicrophoneSpaceUserMessage.userId !== this.userId?.toString()) break;
+                                notificationPlayingStore.playNotification(
+                                    get(LL).notification.askToMuteMicrophone(),
+                                    "audio-mute.svg"
+                                );
+                                requestedMicrophoneState.disableMicrophone();
+                                break;
+                            }
+                            case "muteVideoSpaceUserMessage": {
+                                if (subMessage.muteVideoSpaceUserMessage.userId !== this.userId?.toString()) break;
+                                notificationPlayingStore.playNotification(
+                                    get(LL).notification.askToMuteCamera(),
+                                    "camera-off.png"
+                                );
+                                requestedCameraState.disableWebcam();
+                                break;
+                            }
+                            case "muteMicrophoneEverybodySpaceUserMessage": {
+                                // If the sender receive this message, ignore it
+                                if (
+                                    subMessage.muteMicrophoneEverybodySpaceUserMessage.userId ===
+                                    this.userId?.toString()
+                                )
+                                    break;
+                                notificationPlayingStore.playNotification(
+                                    get(LL).notification.askToMuteMicrophone(),
+                                    "audio-mute.svg"
+                                );
+                                requestedMicrophoneState.disableMicrophone();
+                                break;
+                            }
+                            case "muteVideoEverybodySpaceUserMessage": {
+                                // If the sender receive this message, ignore it
+                                if (subMessage.muteVideoEverybodySpaceUserMessage.userId === this.userId?.toString())
+                                    break;
+                                notificationPlayingStore.playNotification(
+                                    get(LL).notification.askToMuteCamera(),
+                                    "camera-off.png"
+                                );
+                                requestedCameraState.disableWebcam();
+                                break;
+                            }
```

Comment:
Can we apply inversion of control here?
I'm trying to keep the `RoomConnection` ignorant of the rest of the code.
I would need to keep it this way to restore the "benchmark" feature some day.

Ideally, the benchmark feature could use the same "RoomConnection" class as the rest of the code. For this to work, the class must not depend on anything that is related to the front (no svelte store, no direct reference to a service, etc...)
Instead, it should just emit messages that are listened out of the class.

This code should therefore be migrated out of the RoomConnection, inside the GameScene or another dedicated class.